prov/util/src/util_mr_cache.c

```diff
@@ -291,6 +294,14 @@ int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct fi_mr_attr *att
 {
 	struct ofi_mr_info info;
 	int ret;
+	struct ofi_mem_monitor *monitor = cache->monitors[attr->iface];
+
+	if (!monitor) {
+		FI_WARN(&core_prov, FI_LOG_MR,
+			"MR cache disabled for %s memory\n",
+			fi_tostr(&attr->iface, FI_TYPE_HMEM_IFACE));
+		return -FI_EINVAL;
```

Comment:
Can you change this to ENOSYS or ENOTSUPP?

Reply:
Will change to ENOTSUPP.